analysis/cross_partition_combiners.py

```diff
@@ -282,33 +278,59 @@ class CrossPartitionCombiner(pipeline_dp.combiners.Combiner):
     # 1. The sum of non dp metrics, which is used for averaging of error
     # metrics.
     # 2. metrics.UtilityReport contains error metrics.
-    AccumulatorType = Tuple[Tuple, metrics.UtilityReport]
-
-    def __init__(self, dp_metrics: List[pipeline_dp.Metrics],
-                 public_partitions: bool):
+    AccumulatorType = Tuple[Tuple, metrics.UtilityReport, Tuple[float]]
+
+    @staticmethod
+    def partition_contribution_weighing_function(
+            metrics: metrics.PerPartitionMetrics) -> Tuple[float]:
+        return tuple(map(lambda el: el.sum, metrics.metric_errors))
+
+    @staticmethod
+    def equal_weighing_function(
+            metrics: metrics.PerPartitionMetrics) -> Tuple[float]:
+        """
+        Assumes that partition_selection_probability_to_keep for public
+        partitions is 1 and all public partitions including are processed
+        via `create_accumulator`.
+        For the public partitions weights will be 1, and we will do normal
+        averaging because total weight will equal to the total number of
+        partitions. For private partitions we will do weighted average and
+        total weight will equal to mean number of kept partitions
+        (`partitions.kept_partitions.mean`).
+        """
+        return (metrics.partition_selection_probability_to_keep,) * len(
+            metrics.metric_errors)
+
+    def __init__(
+        self,
+        dp_metrics: List[pipeline_dp.Metrics],
+        public_partitions: bool,
+        weighing_function: Callable[[metrics.PerPartitionMetrics],
+                                    Tuple[float]] = equal_weighing_function):
         self._dp_metrics = dp_metrics
         self._public_partitions = public_partitions
+        self.weighing_function = weighing_function
 
     def create_accumulator(
             self, metrics: metrics.PerPartitionMetrics) -> AccumulatorType:
         actual_metrics = tuple(me.sum for me in metrics.metric_errors)
         return actual_metrics, _per_partition_to_utility_report(
-            metrics, self._dp_metrics, self._public_partitions)
+            metrics, self._dp_metrics,
+            self._public_partitions), self.weighing_function(metrics)
 
     def merge_accumulators(self, acc1: AccumulatorType,
                            acc2: AccumulatorType) -> AccumulatorType:
-        sum_actual1, report1 = acc1
-        sum_actual2, report2 = acc2
+        sum_actual1, report1, weight1 = acc1
+        sum_actual2, report2, weight2 = acc2
         sum_actual = tuple(x + y for x, y in zip(sum_actual1, sum_actual2))
         _merge_utility_reports(report1, report2)
-        return sum_actual, report1
+        return sum_actual, report1, tuple(map(sum, zip(weight1, weight2)))
```

Comment:
Nit: extract 
`weight = tuple(map(sum, zip(weight1, weight2)))`

(it would make this impl more homogenous)